### DISCUSSION TITLE
Orange Pi 5 Plus boot from EMMC not working

### DISCUSSION THREAD

**Discussion by thewonderer73** [Joshua-Riek/ubuntu-rockchip#494] [General]
Hi,

I've got my new Orange Pi 5 Plus and EMMC module from OrangePi.  Installed the EMMC module.

I downloaded the latest Orange Pi 5 Plus Ubuntu image from here and imaged my SD card.  Booted successfully.

Then I made a tmp folder and used wget to download the image.

I ran the below commands from the WIKI modifying it for the latest server build.

lsblk /dev/mmcblk0

sudo xz -dc ubuntu-22.04.3-preinstalled-server-arm64-orangepi-5-plus.img.xz | sudo dd of=/dev/mmcblk0 bs=4k

and then

sync

This is how it looks with the SD card in and booting.

 lsblk
NAME         MAJ:MIN RM  SIZE RO TYPE MOUNTPOINTS
mtdblock0     31:0    0   16M  0 disk
mmcblk0      179:0    0 28.9G  0 disk
├─mmcblk0p1  179:1    0  512M  0 part
└─mmcblk0p2  179:2    0  6.4G  0 part
mmcblk0boot0 179:32   0    4M  1 disk
mmcblk0boot1 179:64   0    4M  1 disk
mmcblk1      179:96   0 29.8G  0 disk
├─mmcblk1p1  179:97   0  512M  0 part /boot/firmware
└─mmcblk1p2  179:98   0 29.3G  0 part /

When I remove the SD card, the Orange Pi doesn't boot, the LED just stays blue constantly.  Am I missing something?

Thanks.

**Comment by thewonderer73** [comment]
I tried Armbian and using the armbian-config tool but that didn't work either.

Now going to trying official Orange Pi 5 plus install documentation

**Reply by Joshua-Riek** [reply]
Try to erase the SPI flash with the below command (it may take a few minutes).
```bash
sudo dd if=/dev/zero of=/dev/mmcblk0 count=4096 bs=512
```

**Reply by thewonderer73** [reply]
Thanks.

I ran the command. Restarted.

Ran the previous command to re-image across.

Before restarted I grabbed the current disk/partitation state as below (got Armbian on the SD card at the moment).

Restarted and back to a blue light.  I tried plugging in the HDMI cable.  No luck.

Tried even the orange pi 5 plus guide using the tools and additional USB cable at the front.  All seemed to go well, but once again, solid blue led light once I power back on.  I don't get it.

NAME         MAJ:MIN RM  SIZE RO TYPE MOUNTPOINTS
mtdblock0     31:0    0   16M  0 disk
mmcblk0      179:0    0 28.9G  0 disk
├─mmcblk0p1  179:1    0  512M  0 part
└─mmcblk0p2  179:2    0  6.4G  0 part
mmcblk0boot0 179:32   0    4M  1 disk
mmcblk0boot1 179:64   0    4M  1 disk
mmcblk1      179:96   0 29.8G  0 disk
├─mmcblk1p1  179:97   0  256M  0 part /boot
└─mmcblk1p2  179:98   0 29.2G  0 part /var/log.hdd
                                      /
zram0        254:0    0  1.8G  0 disk [SWAP]
zram1        254:1    0   50M  0 disk /var/log

**Reply by Joshua-Riek** [reply]
I know emmc boot works, ive tested it on 2 of my boards. So i do not know what is wrong here.

Clearing the SPI nor or writing the bootloader to the SPI nor will resolve this problem.

**Reply by thewonderer73** [reply]
Thanks Joshua, appreciate your response.  Appreciate even more, the work you've done bringing Ubuntu to the Orange Pi 5 boards.

I did read some where that some people were modifying a /boot/  txt file and pointing it to a .DTS file to assist with making EMMC boot off the EMMC chip.

I did tried Armbian 23.11. build.  Interesting that the 6.7 kernel draws more power 3 watts vs 2 watts for the 5.10 build .  Suspect it's enabling some other hardware on the device, but just a guess.

For now, I don't have the time to try much more with the EMMC so will stick with a 'fast' SD card for now.  Not sure what the read speed of the SD card hardware is in the OPi 5s .  I know the RPI5 has improved SD card performance.

**Comment by thewonderer73** [comment]
Hi Joshua,

I got it working!

So I had to clear the SPI flash.

I was playing with Armbian OS as the Armbian-Config menu had the option to write to EMMC, which still wasn't working.

Anyway, I had to do a fdisk -l to identify and SPI 16MB chip, then install gdisk to select and clear the partitions on the chip.
Then I ran Armbian-config to install the system to the MTD (SPI) ,wait 5 mins and shutdown.
Then I ran the config again, installed to EMMC and shutdown, removed SD card and restarted.

I've put these details here in case anyone else encounters the issue.

https://www.crosstalksolutions.com/orange-pi-5-simple-overview-and-installation-with-m-2-ssd/

**Comment by sean-sypark** [comment]
It seems a problem of spi flash.
$sudo orangepi-config
System > Install >"Install/Update the bootloader on SPI Flash" OK

After this,  OPi5+ was booted with emmc in my case.

**Reply by assembleer** [reply]
This worked for me. Before I kept erasing spi and writing the OS image to emmc with rkdevtool and never succeeded, until I did install/update the bootloader on SPI flash. (Don't know if it was needed, but also did a bootloader update to emmc.

**Reply by mrbluecoat** [reply]
Updating the bootloader on SPI flash fixed it for me as well. Thanks!

![image](https://github.com/Joshua-Riek/ubuntu-rockchip/assets/382164/c313355e-ff5b-4825-af3c-d1ae1f53c9c6)

**Reply by shayaansiddiqui** [reply]
Do you update the SPI flash before or after installing Ubuntu on the eMMC?

**Reply by assembleer** [reply]
I did it after installation, when I noticed it did not boot from emmc

**Reply by Joshua-Riek** [reply]
Spi does not need to be updated for the emmc to boot. 

https://github.com/Joshua-Riek/ubuntu-rockchip/wiki/Ubuntu-22.04-LTS#install-ubuntu-onto-an-emmc-from-linux

**Reply by assembleer** [reply]
Was necessary in my situation. And for sure flashing the boot sector to spi flash solved my problem of not booting from emmc. So for some doing this might help.

**Comment by thewonderer73** [comment]
No issues imaging with my OPi 3B using recommended commands in the Wiki (used 5+ instructions on for 3B).  Must have been an issue with the OPi 5+

**Comment by thewonderer73** [comment]
I am curious whether a newer version of Ubuntu or Linux BSP kernel will require the SPI to be re-flashed.  I used Armbian config to help me that time, but if I swap over to the possible 24.04 / BSP 6.1 that may be launched in April, would I need to update the SPI boot loader?  Is this only required when a newer Kernel or Uboot is released?  How would this be done from CLI rather than a menu?  Asking these questions for myself and any other users in the future.  Thanks.
